public/scripts/s3-uploads.js

```diff
@@ -1,61 +1,90 @@
-(function () {
-  document.getElementById('camera-input').onchange = function () {
-    const files = document.getElementById('camera-input').files;
+const { saveFileToLocalStorage, addTagAndHeadingToDB, removeItemsFromStorage } = require('./s3-helpers.js');
+
+function s3handling(fileSource) {
```

Comment:
can we export this, then call it in main. (if that makes sense, right now this function gets executed when it is required, but that can be very hard to follow)